bakery-src/tests/test_bakery_scripts.py

```diff
@@ -2472,7 +2531,193 @@ def test_fetch_map_resources(tmp_path, mocker):
         "image_unused.svg",
     ])
 
-    assert(resources_dir.is_dir())
+    assert(initial_resources_dir.is_dir())
+
+
+def test_download_exercise_images(tmp_path, mocker):
+    """Test download exercise images script"""
+    assembled_dir = tmp_path / "assembled"
+    output_dir = tmp_path / "downloaded_exercise"
+    resources_dir = tmp_path / "resources_xyz"
+
+    assembled_dir.mkdir(parents=True)
+    output_dir.mkdir(parents=True)
+    resources_dir.mkdir(parents=True)
+
+    assembled1 = assembled_dir / "assembled1.xhtml"
+    injected_exercise_content = (
+        '<!DOCTYPE html>'
+        '<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US">'
+        '<body>'
+        '<p>Hello! I have an injected exercise with an image on the internet.</p>'
+        '<img src="../resources/testx" />'
+        '<img src="http://127.0.0.1:9999/test.jpg" />'
+        '<img src="../resources/testy" />'
+        '</body>'
+        '</html>'
+    )
+    assembled1.write_text(injected_exercise_content)
+
+    output1 = output_dir / "output1.xhtml"
+
+    # start a local simple http server which emulates serving a JPEG file
+    server = http.server.ThreadingHTTPServer(
```

Comment:
This might not be idiomatic python but what if, rather than starting a webserver in unit tests, the requests were mocked?

Reply:
argh, yes, was not thinking of mocking the http requests.

The test I have now inside (with the simple http server) tests more functionality than mocking the requests. Unsure if I should remove the simple web server. Any other opinions?

Reply:
After discussion with @ckline-tryptic I will keep the current behaviour with the simple http server because it covers more functionality. Nevertheless I will mock http requests in future without writing an entire web server for it!